Sources/SwiftLexicalLookup/LookupResult.swift

```diff
@@ -46,4 +52,54 @@ import SwiftSyntax
       return .fromScope(scope, withNames: names)
     }
   }
+
+  /// Debug description of this lookup name.
+  @_spi(Experimental) public var debugDescription: String {
+    var description =
+      resultKindDebugName + ": " + scope.scopeDebugDescription
+
+    switch self {
+    case .lookInMembers:
+      break
+    default:
+      if !names.isEmpty {
+        description += "\n"
+      }
+    }
+
+    for (index, name) in names.enumerated() {
+      if index + 1 == names.count {
+        description += "`-" + name.debugDescription
+      } else {
+        description += "|-" + name.debugDescription + "\n"
+      }
+    }
+
+    return description
+  }
+
+  /// Debug name of this result kind.
+  private var resultKindDebugName: String {
+    switch self {
+    case .fromScope:
+      return "fromScope"
+    case .fromFileScope:
+      return "fromFileScope"
+    case .lookInMembers:
+      return "lookInMembers"
+    }
+  }
+}
+
+@_spi(Experimental) extension [LookupResult] {
+  /// Debug description this array of lookup results.
+  @_spi(Experimental) public var debugDescription: String {
+    var str: String = ""
+
+    for (index, result) in self.enumerated() {
+      str += result.debugDescription + (index + 1 == self.count ? "" : "\n")
+    }
+
+    return str
```

Comment:
You should be able to use `joined(separator: )` for this

```suggestion
    return self.map(\.debugDescription).joined(separator: "\n")
```